File: scratch/replace_kota.php
<?php

$directories_to_scan = [
    'resources/views',
    'app',
    'database/seeders',
    'public',
    'routes',
    'config'
];
$base_dir = "d:\\PBL\\pbls6";

$replacements = [
    '/\\bKota Pasuruan\\b/' => 'Kabupaten Pasuruan',
    '/\\bkota pasuruan\\b/' => 'kabupaten pasuruan',
    '/\\bKOTA PASURUAN\\b/' => 'KABUPATEN PASURUAN',
    '/\\bPemkot Pasuruan\\b/' => 'Pemkab Pasuruan',
    '/\\bpemkot pasuruan\\b/' => 'pemkab pasuruan',
    '/\\bPemerintah Kota Pasuruan\\b/' => 'Pemerintah Kabupaten Pasuruan',
    '/\\bpemerintah kota pasuruan\\b/' => 'pemerintah kabupaten pasuruan',
    '/\\bpemerintah kota\\b/' => 'pemerintah kabupaten',
    '/\\bPemerintah Kota\\b/' => 'Pemerintah Kabupaten',
    '/Visi Misi Kota/' => 'Visi Misi Kabupaten',
    '/Misi Kota/' => 'Misi Kabupaten',
];

function replace_in_file($filepath, $replacements) {
    global $base_dir;
    
    $content = file_get_contents($filepath);
    $new_content = $content;
    
    foreach ($replacements as $pattern => $replacement) {
        $new_content = preg_replace($pattern, $replacement, $new_content);
    }
    
    if ($new_content !== $content) {
        file_put_contents($filepath, $new_content);
        echo "Updated: " . str_replace($base_dir, '', $filepath) . "\n";
    }
}

function process_directory($dir, $replacements) {
    if (!is_dir($dir)) return;
    
    $iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator($dir));
    foreach ($iterator as $file) {
        if ($file->isFile()) {
            $ext = $file->getExtension();
            if (in_array($ext, ['php', 'js', 'json', 'html', 'env', 'txt'])) {
                replace_in_file($file->getPathname(), $replacements);
            }
        }
    }
}

foreach ($directories_to_scan as $dir) {
    $path = $base_dir . DIRECTORY_SEPARATOR . $dir;
    process_directory($path, $replacements);
}

// Also process environment and related root files
replace_in_file($base_dir . DIRECTORY_SEPARATOR . '.env', $replacements);
replace_in_file($base_dir . DIRECTORY_SEPARATOR . '.env.example', $replacements);

echo "Done\n";

?>

File: scratch/replace_kota.py
import os
import re

directories_to_scan = [
    'resources/views',
    'app',
    'database/seeders',
    'public',
    'routes',
    'config'
]
base_dir = r"d:\PBL\pbls6"

def replace_in_file(filepath):
    try:
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
            
        new_content = content
        
        # Replace variations of Kota Pasuruan
        replacements = [
            (r'\bKota Pasuruan\b', 'Kabupaten Pasuruan'),
            (r'\bkota pasuruan\b', 'kabupaten pasuruan'),
            (r'\bKOTA PASURUAN\b', 'KABUPATEN PASURUAN'),
            (r'\bPemkot Pasuruan\b', 'Pemkab Pasuruan'),
            (r'\bpemkot pasuruan\b', 'pemkab pasuruan'),
            (r'\bPemerintah Kota Pasuruan\b', 'Pemerintah Kabupaten Pasuruan'),
            (r'\bpemerintah kota pasuruan\b', 'pemerintah kabupaten pasuruan'),
            (r'\bpemerintah kota\b', 'pemerintah kabupaten'),
            (r'\bPemerintah Kota\b', 'Pemerintah Kabupaten'),
            (r'Visi Misi Kota', 'Visi Misi Kabupaten'),
            (r'Misi Kota', 'Misi Kabupaten'),
        ]

        for old, new in replacements:
            new_content = re.sub(old, new, new_content)

        if new_content != content:
            with open(filepath, 'w', encoding='utf-8') as f:
                f.write(new_content)
            print(f"Updated: {filepath}")
    except Exception as e:
        print(f"Error processing {filepath}: {e}")

for directory in directories_to_scan:
    dir_path = os.path.join(base_dir, directory)
    if os.path.exists(dir_path):
        for root, dirs, files in os.walk(dir_path):
            for file in files:
                if file.endswith(('.php', '.js', '.json', '.html', '.env', '.txt')):
                    filepath = os.path.join(root, file)
                    replace_in_file(filepath)
